### PR TITLE
Reply By Tweet Debugging

### PR DESCRIPTION
**Pull request by namangupta01**

### CODE REVIEW

**Comment by namangupta01** [comment]
@gauravano can we merge this?

**Comment by namangupta01** [comment]
Restarted travis.

**Comment by gauravano** [comment]
It's failing for all PRs @namangupta01

**Comment by namangupta01** [comment]
Oh okay. So it's not related to mine. We can merge this then?

File: app/models/comment.rb [hunk 1]
```diff
@@ -439,20 +439,58 @@ def self.check_and_add_tweets(tweets)
   end
 
   def self.get_node_from_urls_present(urls)
+    puts ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+    puts "all urls ->"
+    puts urls
+    puts ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
     urls.each do |url|
+      puts ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+      puts "url.include? https:// -> #{url.include? "https://"}"
```
Comment (seabl) [inline app/models/comment.rb]: Maybe you could use the start_with? string method instead of include? !

File: app/models/comment.rb [hunk 2]
```diff
@@ -439,20 +439,58 @@ def self.check_and_add_tweets(tweets)
...
+      puts ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
       next unless url.include? "https://"
 
       if url.last == "."
         url = url[0...url.length - 1]
       end
+
```
Comment (seabl) [inline app/models/comment.rb]: And here (lines 452 and 453) you could use the **end_with?** and **chomp** methods

**Comment by namangupta01** [comment]
Hi @seabl , this is just for debugging. After we are done with the debugging we are going to remove it.
@jywarren can we merge this real quick so that we can further investigate what the actual problem is?
Thanks

File: app/models/comment.rb (hunk 2 above)
Comment (jywarren) [reply app/models/comment.rb]: Thanks for the review, @seabl!

**Comment by jywarren** [comment]
Hi @namangupta01 i'm afraid I'd prefer to not merge a failing build, as it can mask other errors being merged. Can you try helping to solve #6094 do you think, as this is an urgent situation for all contributors? 

Alternatively, is there really no way to test this happening in your dev environment or in unstable? Perhaps @SidharthBansal may be able to suggest a clever way to test it. Just to be clear, is it possible to get working locally in production mode, or in unstable? Thanks!

**Comment by namangupta01** [comment]
I only have a couple of concern while testing on `unstable` which is `unstable` donot have upto date database to sync with the notes mention in tweets and if there are I don't know which one it is. And another concern is does login via twitter works on unstable.

**Comment by namangupta01** [comment]
Sure I will have look at #6094 and try to solve it along with you guys.
Thanks

**Comment by jywarren** [comment]
In the original issue https://github.com/publiclab/plots2/issues/4962 (i
think) I mentioned some specific tweets you can use. The other option is to
try pointing at a different Twitter account on unstable, like your own, and
to test there using a new user account on unstable, and posting a new test
note.

**Comment by namangupta01** [comment]
Hi, I tried this on local with a tweet from one of my account and replied to that tweet from another account and I was able to find the node from that tweet but in Production, I am not able to. I need a couple of information to get it fix and running it on production. First is what is the environment variable `ENV["WEBSITE_HOST_PATTERN"]` setted in there and second is what is the data we store in UserTag when we do login via twitter -- I need this just to confirm that everything will work fine on production because I am not able to run login via twitter on local and get that value. I guess these are the small things that we have to look into it. Afterwards, it will be good to run. 
Also having discussion at https://github.com/publiclab/plots2/issues/6085
Thanks

**Comment by jywarren** [comment]
OK, i'll pull those data now!

WEBSITE_HOST_PATTERN=://publiclab.org/n/

For my twitter oauth i got: `oauth:twitter:15024966` - you can see it on my
profile page!

**Comment by namangupta01** [comment]
Thanks! Is this also same in stable.publiclab.org i.e WEBSITE_HOST_PATTERN=://publiclab.org/n/
Because I am downloading logs from http://stable.publiclab.org/cron_log1.log

**Comment by jywarren** [comment]
Yes, it is the case there too!

**Comment by namangupta01** [comment]
It is weird. When I set the same on local I get the node id extracted from the link.

**Comment by namangupta01** [comment]
Can we also merge this? Since I have tested on local and it's able to identify the node_id easily which it is not getting on production on checking logs? What do you think?

**Comment by jywarren** [comment]
Yes! No danger that we'll fill up logs too fast or anything else to watch out for? Thanks!

**Comment by jywarren** [comment]
I guess it's fine. But please open up an issue pointing to this PR so that we remember to take out these comments too! Thanks!

**Comment by jywarren** [comment]
OK, i hope to publish today but not sure. Thanks!

**Comment by jywarren** [comment]
I'll leave a note in the chatroom if we manage to!

**Comment by namangupta01** [comment]
Thanks!!! I will create a new issue for removing these comments so that we know we have to remove these comments.

**Comment by namangupta01** [comment]
Hoping this could give us enough information to find out what is wrong exactly. Otherwise, we have to dig for more logs.

**Pull request merged**